src/firefly/js/charts/ui/PlotlyChartArea.jsx

```diff
@@ -1,19 +1,117 @@
-import React, {Component} from 'react';
+import React, {useEffect, useRef} from 'react';
 import PropTypes from 'prop-types';
-import {flux} from '../../core/ReduxFlux.js';
 import {get, set, cloneDeep} from 'lodash';
-import shallowequal from 'shallowequal';
 import {PlotlyWrapper} from './PlotlyWrapper.jsx';
 import {showInfoPopup} from '../../ui/PopupUtil.jsx';
-import {getTheme} from '../../ui/FireflyRoot.jsx';
+import {getTheme, useColorMode} from '../../ui/FireflyRoot.jsx';
 
 import {dispatchChartHighlighted, dispatchChartUpdate, dispatchSetActiveTrace, getAnnotations, getChartData, usePlotlyReact} from '../ChartsCntlr.js';
 import {clearChartConn, flattenAnnotations, handleTableSourceConnections, isSpectralOrder, isScatter2d} from '../ChartUtil.js';
+import {useStoreConnector} from 'firefly/ui/SimpleComponent.jsx';
+import {Skeleton} from '@mui/joy';
 
 const X_TICKLBL_PX = 60;
 const TITLE_PX = 30;
 const MIN_MARGIN_PX = 30; // should be enough to accommodate upper limit annotation
 
+
+export function PlotlyChartArea({chartId, widthPx, heightPx, thumbnail}) {
+
+    useColorMode();     // renders when colorMode changes
+
+    const {data=[], isLoading, highlighted, selected, layout={}, activeTrace=0, xyratio, stretch} = useStoreConnector(() => getChartState(chartId), [chartId]);
+
+    const prevDim = useRef();       // save previous dimension to detect resizing
+    useEffect(() => {
+        prevDim.current = {widthPx, heightPx};
+    }, [widthPx, heightPx]);
```

Comment:
Do you need an effect for this? I think you set just set it on the ref directly at the end of every render.

Reply:
You are right.  I removed it.